### ISSUE TITLE
Add Support to delete current Node Connections / ss-server-addr

### ISSUE THREAD

**Issue by csiggydev** [joel74/POSH-LTM-Rest#171]
Hi Joel,

Great work on the module, works great. I use this script for deployments, disabling nodes, re-enabling them and it would be really useful to kill off the connections after disabling each node. This would have the same effect as TMSH equivalent:

delete /sys connection ss-server-addr 1.1.1.1
delete /sys connection ss-server-addr 11.11.11.11
delete /sys connection ss-server-addr 111.111.111.111

etc.

perhaps a Delete-Connection or Kill-Connection Function?

Thanks again.

**Comment by joel74** [comment]
Hi,
Let me look into that. I think that the approved verb would be clear or remove, but the effect would be the same. Thanks for suggesting this.

**Comment by csiggydev** [comment]
Hey Joel,

Well, the node is at the parent level higher than a pool-member in a pool to which it may also be associated. So, if you're disabling a given node, which is also serving as a pool-member in a particular pool, any pool member in the configuration that uses the same IP address is also set to 'Disabled' or 'Forced Offline.' This sounds like a mere announcement from BIG-IP, and yes, it's expected behavior. More importantly, did the function allow you to actually delete the existing connection successfully, despite the message?

Thanks for your work on this one.

**Comment by joel74** [comment]
No, the connections to the node persist and aren't deleted. The same message is given, regardless of if the node is disabled, forced offline, or enabled.

**Comment by Matt-Chambers-FTR** [comment]
Joel, I was wondering if you've ever made any progress on the Delete-Connection or Kill-Connection Function?
I've been writing some scripts to monitor a few of our more active pools.  I've tried adding some logic to verify I'm pulling stats from the active device and not the standby.  But my session always seems to stay with the first device I make a connection into. 
#My Creds
$pwstring = ‘mypassword’
$secpasswd = ConvertTo-SecureString $pwstring -AsPlainText -Force
$LTMCredentials = New-Object System.Management.Automation.PSCredential "myID", $secpasswd

#DEPENDENCY
Import-Module F5-LTM -Force -ErrorAction Stop

#Find which LB is Active
$F5Session = New-F5Session -LTMName $IP-of-first-LB -LTMCredentials $LTMCredentials 
$status = Get-F5Status
$status
if ($status -ne "ACTIVE")
    {
        echo "1st LB not online, will check 2nd LB"
        $F5Session = New-F5Session -LTMName $IP-of-second-LB -LTMCredentials $LTMCredentials 
        $status = Get-F5Status 
        $status
    } 

Even if the second LB is active, the results of this show
STANDBY
1st LB not online, will check 2nd LB
STANDBY 

I do know the second LB is online.  If I reverse which IP is checked first it does show me it’s active.
ACTIVE
1st LB online 

I’m hoping that if I could drop any connections to the first I’d then be able to successfully establish a session with the second LB.

**Comment by joel74** [comment]
Hi,

A couple comments. First, there's a quick solution to your issue: when creating a new session, use the -PassThru switch.

I.e.: 
$F5_session = New-F5Session -LTMName 192.168.1.1 -LTMCredentials $credsA -Default -PassThru
Without that, you're not saving the session object, and calling Get-F5Status just uses the default session. Also, Get-F5Status takes a -Session param, so you'll need to pass that in both times.

Secondly, sessions are different from connections. It seemed like you might be confusing the two. Sessions are for working with the LTM API and connections are browsers or clients that are using LTM pool / node resources. I hope that helps.

**Comment by csiggydev** [comment]
Sorry for the long delay on this. @joel74 - Could you try your suggested verbs from above, but this time to kill an existing connection to a given pool-member? This would at least achieve almost the desired result. In our topology, we don't use the same node in multiple pools, and perhaps pool-member associated verbs are more applicable.

**Comment by Obiron801** [comment]
@csiggydev I just came across this issue and, if you're still looking for an answer, I've implemented my own in conjunction with joel's scripts.

There are two components to how I'm doing this. First, I never found a way to run the delete commands directly from the api. But you can create a script in the F5 with those commands, and then call that script via the api.
I create one script to delete connections for a whole node, and another script to delete connections for a single node/port combo

In tmsh:
create cli script delconnnode.tcl
Edit the run section to look like this:
`proc script::run {} {
        tmsh::delete sys connection ss-server-addr [lindex $tmsh::argv 1]
    }`

create cli script delconnpool.tcl
Edit the run section to look like this:
`proc script::run {} {
        tmsh::delete sys connection ss-server-addr [lindex $tmsh::argv 1] ss-server-port [lindex $tmsh::argv 2]
    }`


With those scripts living on the F5, here's the function I created to call the scripts (this could probably be cleaned up a bit using the $F5Session, but I had a lot of this written before I came across joel's scripts):
`Function Clear-SysConnection()
{
	param($serverAddr,$port)
	
	If (!$port) {
		$JSONBodyScript = @{command="run";utilCmdArgs="delconnnode.tcl $serverAddr"} | ConvertTo-Json
	}
	Else {
		$JSONBodyScript = @{command="run";utilCmdArgs="delconnpool.tcl $serverAddr $port"} | ConvertTo-Json
	}
	
	$JSONBodyToken = [pscustomobject]@{
		username=$lbAdminUser
		password=$lbAdminPassword
		loginProviderName="tmos"
	} | ConvertTo-Json
	
	$tokenString = Invoke-WebRequest -ContentType "application/json" -Method POST -Uri https://$lbAddress/mgmt/shared/authn/login -Body $JSONBodyToken -SkipCertificateCheck | Select-String -Pattern "token"
	
	$SearchStart = "`"token`":`""
	$SearchEnd = "`",`"name`":"
	$tokenString -match "(?s)$SearchStart(?<content>.*)$SearchEnd" | Out-Null
	$token=$matches['content']
	
	$Headers = @{'X-F5-Auth-Token' = $token}
	
	Invoke-WebRequest -Headers $Headers -ContentType "application/json" -Method POST -Uri https://$lbAddress/mgmt/tm/cli/script -Body $JSONBodyScript -SkipCertificateCheck | Out-Null
	
}`

I hope this helps!

**Comment by joel74** [comment]
I'm going to close this issue. @Obiron801 , I appreciate the work-around you offered a while back. Since that solution requires scripts to live locally on the F5, it's outside the scope of what this module can accomplish. Great info to have, though.
